src/lib/Wst/Offchain/Endpoints/Deployment.hs

```diff
@@ -0,0 +1,196 @@
+{-| Deploy the directory and global params
+-}
+module Wst.Offchain.Endpoints.Deployment(
+  deployTx,
+  deployBlacklistTx,
+  insertNodeTx,
+  issueProgrammableTokenTx,
+  issueSmartTokensTx,
+  transferSmartTokensTx,
+  insertBlacklistNodeTx,
+  blacklistCredentialTx,
+  seizeCredentialAssetsTx,
+) where
+
+import Cardano.Api (Quantity)
+import Cardano.Api.Shelley qualified as C
+import Control.Monad (when)
+import Control.Monad.Except (MonadError)
+import Control.Monad.Reader (MonadReader, asks)
+import Convex.BuildTx (payToAddress)
+import Convex.BuildTx qualified as BuildTx
+import Convex.Class (MonadBlockchain, MonadUtxoQuery)
+import Convex.CoinSelection qualified
+import Data.Foldable (maximumBy)
+import Data.Function (on)
+import SmartTokens.Types.PTokenDirectory (DirectorySetNode (..))
+import Wst.AppError (AppError)
+import Wst.Offchain.BuildTx.DirectorySet (InsertNodeArgs (inaNewKey))
+import Wst.Offchain.BuildTx.DirectorySet qualified as BuildTx
+import Wst.Offchain.BuildTx.ProgrammableLogic qualified as BuildTx
+import Wst.Offchain.BuildTx.ProtocolParams qualified as BuildTx
+import Wst.Offchain.BuildTx.TransferLogic qualified as BuildTx
+import Wst.Offchain.BuildTx.TransferLogic qualified as TL
+import Wst.Offchain.Env qualified as Env
+import Wst.Offchain.Query (UTxODat (..))
+import Wst.Offchain.Query qualified as Query
+
+{-| Build a transaction that deploys the directory and global params. Returns the
+transaction and the 'TxIn' that was selected for the one-shot NFTs.
+-}
+deployTx :: (MonadReader env m, Env.HasOperatorEnv era env, MonadBlockchain era m, MonadError (AppError era) m, C.IsBabbageBasedEra era, C.HasScriptLanguageInEra C.PlutusScriptV3 era) => m (C.Tx era, C.TxIn)
+deployTx = do
+  (txi, _) <- Env.selectOperatorOutput
+  opEnv <- asks Env.operatorEnv
+  (tx, _) <- Env.withEnv $ Env.withOperator opEnv $ Env.withDirectoryFor txi
+              $ Env.balanceTxEnv_
+              $ BuildTx.mintProtocolParams >> BuildTx.initDirectorySet
+  pure (Convex.CoinSelection.signBalancedTxBody [] tx, txi)
+
+{-| Build a transaction that inserts a node into the directory
+-}
+insertNodeTx :: forall era env m. (MonadReader env m, Env.HasOperatorEnv era env, Env.HasDirectoryEnv env, MonadBlockchain era m, MonadError (AppError era) m, C.IsBabbageBasedEra era, C.HasScriptLanguageInEra C.PlutusScriptV3 era, MonadUtxoQuery m) => InsertNodeArgs -> m (C.Tx era)
+insertNodeTx args = do
+  -- 1. Find the head node
+  directoryList <- Query.registryNodes @era
+  -- FIXME: Error handling. And how can we actually identify the head node if the query returns more than one?
```

Comment:
Query should never return more than one, since it should query for the utxo with the following NFT:

policyId = directoryNodePolicyId
tokenName = empty currency symbol

The uniqueness of this NFT is enforced by the directory minting policy.